### PR TITLE
Fixed #2439

### LINKED ISSUE #2439: PyroMining and Slimefun Incompatibility
## :round_pushpin: Description (REQUIRED)

Whilst using the PyroMining plugin along side Slimefun's Explosive Pickaxe, it appears that players are able to mass produce the items the plugin drops from mining the ore. I've gone ahead and talked to Pyro (developer of the PyroMining plugin), and he's mentioned that he has already set the events to the highest priority so that its more than likely Slimefun overriding something.

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)

https://youtu.be/3wgpkvai008

## :bulb: Expected behavior (REQUIRED)

Prevent the Explosive Pickaxe from being able to override the events from other plugins.

## :compass: Environment (REQUIRED)

 - Server Software (Spigot/Paper):
Capture

### PR DESCRIPTION
**Pull request by LinoxGH**
## Description

Added a check if the event is cancelled to explosive tool.
Changed BlockBreakEvent and BlockPlaceEvent listener priorities to HIGHEST from MONITOR.

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Review by svr333 — DISMISSED** [review]
Dont exactly know why we'd have to change from Monitor to highest, but changes look good to me

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java [hunk 1]
```diff
@@ -52,6 +52,8 @@ public ExplosiveTool(Category category, SlimefunItemStack item, RecipeType recip
     @Override
     public ToolUseHandler getItemHandler() {
         return (e, tool, fortune, drops) -> {
+            if (e.isCancelled()) return;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java:55]: This is completely redundant as the Listener which calls this method already checks if it is cancelled.
Also missing brackets.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java [hunk 2]
```diff
@@ -75,7 +75,7 @@ public void onBlockPlace(BlockPlaceEvent e) {
         }
     }
 
-    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java:78]:
```suggestion
    @EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
```

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java (hunk 2 above)
Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java:78]: Can I ask why were even changing this?

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java:78]: The bug was about this running AFTER HIGHEST, which should not be done anyway. By moving it TO HIGHEST we basically didn't fix the issue but rather let randomness decide.

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java:78]: In theory events should try to run on NORMAL unless there is a valid reason not to, due to this being dependent on some things to process first, there is a point to be made to move it to HIGH but HIGHEST is definitely too late.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java (hunk 1 above)
Comment (LinoxGH) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java:55]: > This is completely redundant as the Listener which calls this method already checks if it is cancelled.
> Also missing brackets.

No it doesn't

Comment (LinoxGH) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java:55]: It only does a material check.

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java:55]: **ignoreCancelled = true**
```java
@EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
    public void onBlockBreak(BlockBreakEvent e) {
```

Comment (LinoxGH) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java:55]: > **ignoreCancelled = true**
> ```java
> @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
>     public void onBlockBreak(BlockBreakEvent e) {
> ```

Yes. So listener is fired no matter if the even is cancelled. If that worked like it ignored cancelled events, then why is there a cancelled check below?

Comment (LinoxGH) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java:55]: Ok just looked up, I was wrong.

**Review by TheBusyBiscuit — APPROVED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java (hunk 1 above)
Comment (svr333) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java]: ignoreCancelled is very ambiguous

**Review by svr333 — APPROVED** [review]

**Pull request merged**